telemetry/src/main/java/datadog/telemetry/TelemetrySystem.java

```diff
@@ -37,12 +39,15 @@ static Thread createTelemetryRunnable(
       OkHttpClient okHttpClient,
       DependencyService dependencyService) {
     DEPENDENCY_SERVICE = dependencyService;
+
+    List<TelemetryPeriodicAction> actions = new ArrayList<>();
+    actions.add(new IntegrationPeriodicAction());
+    if (Config.get().isTelemetryDependencyServiceEnabled()) {
```

Comment:
```suggestion
    if (null != dependencyService) {
```

This avoids checking the same flag twice and also avoids a NPE when the tracer jar is used to send sample traces:
```
$ java -jar dd-java-agent/build/libs/dd-java-agent-1.4.0-SNAPSHOT.jar sampleTrace
[dd.trace 2023-01-06 00:11:29:212 +0000] [main] INFO com.datadog.appsec.AppSecSystem - AppSec is ENABLED_INACTIVE with powerwaf(libddwaf: 1.5.1) no rules loaded
[dd.trace 2023-01-06 00:11:29:239 +0000] [dd-telemetry] ERROR datadog.telemetry.TelemetryRunnable - Uncaught exception in dd-telemetry
java.lang.NullPointerException
	at datadog.telemetry.dependency.DependencyPeriodicAction.doIteration(DependencyPeriodicAction.java:17)
	at datadog.telemetry.TelemetryRunnable.run(TelemetryRunnable.java:54)
	at java.lang.Thread.run(Thread.java:750)
[dd.trace 2023-01-06 00:11:29:652 +0000] [dd-task-scheduler] INFO datadog.trace.agent.core.StatusLogger - DATADOG TRACER CONFIGURATION {...}
... completed 6 traces
```

Reply:
Done